experimental/PropertyDDS/packages/property-dds/src/test/propertyTree.spec.ts

```diff
@@ -12,15 +12,150 @@ import { requestFluidObject } from "@fluidframework/runtime-utils";
+
+interface Result {
+    container: IContainer;
+    client: SharedPropertyTree;
+}
+
+interface withSummarizer extends Result {
+    summarizer: Awaited<ReturnType<typeof createSummarizer>>;
+}
+
+describe("PropertyDDS summarizer", () => {
+    let objProvider: ITestObjectProvider;
+    const propertyDdsId = "PropertyTree";
+    const USERS = "users";
+    const getClient = async (withSummarizer = false, load = false) => {
+        const container = await (load ? objProvider.loadTestContainer() : objProvider.makeTestContainer());
+        const dataObject = await requestFluidObject<ITestFluidObject>(container, "/");
+
+        let summarizer;
+        if (withSummarizer) {
+            summarizer = await getSummarizer(container);
+        }
+        const client = await dataObject.getSharedObject<SharedPropertyTree>(propertyDdsId);
+        const res: withSummarizer = {
+            summarizer,
+            client,
+            container
+        };
+        return res;
+    };
+
+    const getSummarizer = async (container) => {
+        const summarizer = await createSummarizer(objProvider, container);
+        return summarizer;
+    };
+
+    const createUserNode = (name: string) => {
+        const node = PropertyFactory.create<NodeProperty>("NodeProperty");
+        node.insert("name", PropertyFactory.create("String", undefined, name));
+    };
+
+    beforeEach(async () => {
+        const driver = new LocalServerTestDriver();
+        const registry = [[propertyDdsId, new PropertyTreeFactory()]] as ChannelFactoryRegistry;
+
+        objProvider = new TestObjectProvider(
+            ContainerLoader as any,
+            driver,
+            () =>
+                new TestContainerRuntimeFactory(
+                    "@fluid-experimental/test-propertyTree",
+                    new TestFluidObjectFactory(registry)
+                )
+        );
+
+    });
+
+    it("Scenario 1", async function() {
+        /**
+         * This test produces a scenario where a summarizer creates a summary with an empty remoteChanges array.
+         * This can happen, if the summarizer client prunes all remote changes. However, all remote changes will only be
+         * pruned, when the last operation in the stream is not a changeset op (e.g. a join/remove operation).
+         * In addition to that, the minimum sequence number has to point to the last operation in the stream. For this,
+         * all clients must be synced and have updated their referenceSequenceNumber.
+         *
+         * We create this scenario to reproduce a bug, where a client that joined after such a summary had been created,
+         * submitted an operation with an incorrect referenceGuid. Clients that already had been joined to the session
+         * before the summarization, then performed an incorrect rebase which resulted in exceptions or an incorrect
+         * state in the property tree after the rebase.
+         */
+        this.timeout(30000);
```

Comment:
@DLehenbauer Is there  a better way to ensure that all clients have been updated to the newest sequence number? Waiting is not very robust in a test.

Reply:
@Abe27342 do you have advice here?

Reply:
(that was my assumption, too)

Reply:
based on the doc comment explaining what the test is doing, it needs to advance the MSN to the point of all clients being basically up-to-date. ensureSynchronized isn't sufficient for this. Seems like there might be some code in container runtime to cause immediate noop sending on quorum proposals (to get speedy acceptance), but I've never used that. Ideally our smaller set of mocks would support the scenario you're trying to write, but I'm not sure they do based on the comment. At the very least you should be able to reduce the no-op sending time using client configuration, but you could also consider modifying your data object definition and submitting ops to other DDSes on each client (so that no-ops aren't necessary)

Reply:
I see.  'ensureSynchronized' ensures the clients have received and processed all messages, but the service doesn't know how far a client has advanced until it observes the refSeq of the next op sent by that client, the MSN is always behind immediately following 'ensureSynchronized'.

I like Abe's suggestion of writing to a second DDS on each each client and synchronizing again to push the MSN past the last property DDS op.

Reply:
> I like Abe's suggestion of writing to a second DDS on each each client and synchronizing again to push the MSN past the last property DDS op.

Yes, I think that should work. I can give it a try.

Reply:
Ok, I did change the test to perform a modification of another DDS to force the updates. This work, and we no longer need to wait for such a long time. It still takes 2s because that is the default interval for the noops to be sent and we have to wait for the summarizer to send its noop.

I dont't think there is a way to reduce this time via the official API. If we would directly access the internal collabWindowTracker, we could for the submission of a noop, by calling its submitNoop Api.

Reply:
@DLehenbauer Could we merge this PR? It is a rather important bugfix.